### PR TITLE
added POST method to /users endpoint

### PR DESCRIPTION
**Pull request by EANimesha**
fixes: #37 
**request :** 
 endpoint : /users (POST)
 payload: 
  

>  Content-type- json
>   {
>     "first_name":"Name1",
>     "last_name":"Name2",
>     "email":"n@gmail.com"
>  }

**response:**

> {
>     "timestamp": 1615792926283,
>     "data": {
>         "_id": "604f0b1e4e98a0516137b359",
>         "first_name": "Name1",
>         "last_name": "Name2",
>         "email": "n@gmail.com"
>     }
> }

### CODE REVIEW

**Comment by adityabisoi** [comment]
@EANimesha We do not need to include the "data"  and "_id" fields. The timestamp should be the last field in proper format, `MM-DD-YY-HH-MM`
```
"first_name":
"last_name":
"email":
"timestamp":
```

**Comment by EANimesha** [comment]
Updated. now response is 
```
{
    "first_name": "Name1",
    "last_name": "Name2",
    "email": "n@gmail.com",
    "timestamp": "2021-03-18 19-44"
}
```

**Comment by adityabisoi** [comment]
This PR still needs some corrections. Firstly, there is no need to create a mongodb instance from the accepted data. Also,
the user might send data with fields other than `first_name`, `last_name`, and `email`. The issue asks to send `any` data that is received, back with a timestamp.

File: src/api/routes/users.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import express, { Application, Request, Response, NextFunction } from 'express'
-import mongoose from 'mongoose'
+var dateFormat = require('dateformat');
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:2]: Change this to `import` statement

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

File: src/api/routes/users.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import express, { Application, Request, Response, NextFunction } from 'express'
-import mongoose from 'mongoose'
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:2]: Why was this line removed?

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Review by ShauryaAg — APPROVED** [review]

**Comment by ShauryaAg** [comment]
@EANimesha Can you please resolve the merge conflicts?

**Comment by EANimesha** [comment]
> @EANimesha Can you please resolve the merge conflicts?

@ShauryaAg resolved conflicts

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@EANimesha The POSTed user object should properly fit into the user model and should be checked for the presence and validity of all fields a user must have. Please update the logic to do that.

**Comment by paramsiddharth** [comment]
@EANimesha It would be appropriate to validate the schema before successfully sending back the response with status 200. Extra/Missing parameters should be checked for and errors should be shown accordingly.

File: src/api/routes/users.ts [hunk 3]
```diff
@@ -47,6 +47,13 @@ router.get('/:userId', (req: Request, res: Response, next: NextFunction) => {
         })
 })
 
+
+router.post('/', (req: Request, res: Response, next: NextFunction) => {
+    var userData=req.body
+    userData['timestamp']=dateFormat(new Date(), "yyyy-mm-dd HH-MM")
+    res.status(200).json(userData)
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:54]: @EANimesha Can you please update the PR to verify the data sent with the user schema. `req.body` should not contain any field that is not used by the `user` model.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Comment by adityabisoi** [comment]
@EANimesha This issue has been laying around for a while now. Please make the commented changes and I shall merge it.

**Comment by EANimesha** [comment]
Sorry that branch of my forked repo got deleted. I couldn't open it again. I had to open another PR for that

**Pull request closed without merging**